tests/unit/amo/components/TestCategories.js

```diff
@@ -1,160 +1,100 @@
+import { shallow } from 'enzyme';
 import React from 'react';
-import { findDOMNode } from 'react-dom';
+
 import {
-  renderIntoDocument,
-  findRenderedComponentWithType,
-} from 'react-addons-test-utils';
-import { Provider } from 'react-redux';
-
-import { setViewContext } from 'amo/actions/viewContext';
-import createStore from 'amo/store';
-import { CategoriesBase, mapStateToProps } from 'amo/components/Categories';
-import { setClientApp, setLang } from 'core/actions';
+  CategoriesBase,
+  mapStateToProps,
+} from 'amo/components/Categories';
 import { categoriesLoad } from 'core/actions/categories';
-import { ADDON_TYPE_EXTENSION, CLIENT_APP_ANDROID } from 'core/constants';
+import { ADDON_TYPE_EXTENSION } from 'core/constants';
+import Button from 'ui/components/Button';
+import LoadingText from 'ui/components/LoadingText';
+import { dispatchClientMetadata } from 'tests/unit/amo/helpers';
 import { getFakeI18nInst } from 'tests/unit/helpers';
 
 
-const categoriesResponse = {
-  result: [
-    {
-      application: 'android',
-      name: 'Games',
-      slug: 'Games',
-      type: ADDON_TYPE_EXTENSION,
-    },
-    {
-      application: 'android',
-      name: 'Travel',
-      slug: 'travel',
-      type: ADDON_TYPE_EXTENSION,
-    },
-  ],
-};
-
-describe('Categories', () => {
+describe('<Categories />', () => {
   function render({ ...props }) {
-    const { store } = createStore();
-    store.dispatch(setClientApp('android'));
-    store.dispatch(setLang('fr'));
-    store.dispatch(categoriesLoad(categoriesResponse));
-
-    const { categories } = store.getState().categories;
-    const baseProps = {
-      clientApp: store.getState().api.clientApp,
-      categories: categories[CLIENT_APP_ANDROID][ADDON_TYPE_EXTENSION],
-      dispatch: sinon.stub(),
-    };
-
-    return findRenderedComponentWithType(renderIntoDocument(
-      <Provider store={store}>
-        <CategoriesBase i18n={getFakeI18nInst()} {...baseProps} {...props} />
-      </Provider>
-    ), CategoriesBase);
-  }
-
-  function renderDomNode(props) {
-    return findDOMNode(render(props));
-  }
-
-  it('dispatches setViewContext with addonType', () => {
     const fakeDispatch = sinon.stub();
-    const root = render({
-      addonType: ADDON_TYPE_EXTENSION,
-      dispatch: fakeDispatch,
-    });
 
-    sinon.assert.calledWith(
-      fakeDispatch, setViewContext(ADDON_TYPE_EXTENSION));
-
-    // Make sure that we update the addonType when `componentDidUpdate()`
```

Comment:
Yup, now we use enzyme's `setProps` which calls `componentDidUpdate()` as part of the React Component Lifecycle rather than us calling the method directly, which was what felt off about that code: https://github.com/mozilla/addons-frontend/pull/2807/files/118b437e8e627d07b27d72b0cdf222f6a14c95c1#diff-08275f1fe3a947e23a90f5ae53d3e3c3R63